pkg/controller/constraint/stats_reporter_test.go

```diff
@@ -4,53 +4,116 @@ import (
 	"context"
 	"testing"
 
+	"github.com/open-policy-agent/gatekeeper/v3/pkg/metrics"
 	"github.com/open-policy-agent/gatekeeper/v3/pkg/util"
-	"go.opencensus.io/stats/view"
+	"github.com/stretchr/testify/assert"
+	"go.opentelemetry.io/otel/attribute"
+	sdkmetric "go.opentelemetry.io/otel/sdk/metric"
+	"go.opentelemetry.io/otel/sdk/metric/metricdata"
+	"go.opentelemetry.io/otel/sdk/metric/metricdata/metricdatatest"
 )
 
-func TestReportConstraints(t *testing.T) {
-	const expectedValue int64 = 10
-	const expectedRowLength = 1
-	expectedTags := tags{
-		enforcementAction: util.Deny,
-	}
+type fnExporter struct {
```

Comment:
It seems like this code gets copy/pasted multiple times, would it be worth putting this in a test helper package under /metrics?